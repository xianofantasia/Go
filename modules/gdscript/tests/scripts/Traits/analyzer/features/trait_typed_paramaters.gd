trait SomeTrait:
	pass

class SomeClass:
	uses SomeTrait

signal emits_trait_type(param: SomeTrait)

func receive_trait_type(param: SomeTrait):
	print(param != null)

func test():
	emits_trait_type.connect(receive_trait_type)
	emits_trait_type.emit(SomeClass.new())
	receive_trait_type(SomeClass.new())
	print("ok")
