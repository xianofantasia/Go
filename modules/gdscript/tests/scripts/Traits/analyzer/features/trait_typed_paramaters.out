GDTEST_OK
~~ WARNING at line 13: (RETURN_VALUE_DISCARDED) The function "connect()" returns a value that will be discarded if not used.
~~ WARNING at line 15: (UNSAFE_CALL_ARGUMENT) The argument 1 of the function "receive_trait_type()" requires the subtype "SomeTrait" but the supertype "SomeClass" was provided.
true
true
ok
